src/Microsoft.AspNetCore.ResponseCaching/ResponseCache/IResponseCacheEntry.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.ResponseCaching
+{
+    public interface IResponseCacheEntry
+    {
+        string Key { get; set; }
+        ResponseCacheEntryType Type { get; }
```

Comment:
Not necessarily needed since you can check the types by `<var> is <Type>` and serialize/deserialize with a special value.